cmd/lk/loadtest.go

```diff
@@ -98,18 +98,18 @@ var LoadTestCommands = []*cli.Command{
 	},
 }
 
-func loadTest(cCtx *cli.Context) error {
-	pc, err := loadProjectDetails(cCtx)
+func loadTest(ctx context.Context, cmd *cli.Command) error {
+	pc, err := loadProjectDetails(cmd)
 	if err != nil {
 		return err
 	}
 
-	if !cCtx.Bool("verbose") {
+	if !cmd.Bool("verbose") {
 		lksdk.SetLogger(logger.LogRLogger(logr.Discard()))
 	}
 	_ = raiseULimit()
 
-	ctx, cancel := context.WithCancel(cCtx.Context)
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
This should probably use [NotifyContext](https://pkg.go.dev/os/signal@go1.22.4#NotifyContext).

Reply:
Actually, is this wrapped context even needed? The ancestor context at the root of the app is now a `NotifyContext`, and it looks like the purpose of this code and the following lines is to do the same thing. It is already going to get cleaned up when the program receives these signals.

Reply:
You are right! It does handle more signals, but I don't think it matters for CLI.

Maybe worth doing `git blame` and ask the author why these signals are handled explicitly here.

Reply:
Paging @davidzhao

Reply:
I think I see what it was intending to do, but it doesn't really do that currently. In the release build, If you try to Ctrl-C while the load test is being set up, it's ignored:

```
tobiasfried@slate livekit-cli % livekit-cli load-test --duration 10m
Using default project local
Starting load test with 1 video publishers, 1 subscribers, room: testroom467
^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C im trying to kill itcould not connect Sub 0: could not dial signal connection
could not connect Pub 0: could not dial signal connection
Finished connecting to room, waiting 10m0s

Sub 0 | Track | Kind | Pkts | Bitrate | Dropped

Summary | Tester | Tracks | Bitrate               | Total Dropped | Error
        | Sub 0  | 0/1    | NaNmbps               |  -            | could not dial signal connection
        | Total  | 0/1    | NaNmbps (NaNmbps avg) |  -            | 1
tobiasfried@slate livekit-cli %
```
vs now

```
tobiasfried@slate livekit-cli % bin/lk load-test --duration 10m
Using default project local
Starting load test with 1 video publishers, 1 subscribers, room: testroom851
^C
tobiasfried@slate livekit-cli %
```

Reply:
Looks like the separate context in the original code is there to allow the load tester a chance to print the summary before the program terminates after a SIGINT. It relies on the fact that the root context was *not* hooking these signals, and that the contexts would unwind in reverse.

New code accomplishes the same thing with a single context, so I'm pretty sure its OK?

Reply:
Yeah, I was mostly wondering about the additional signals. Apart from that it should be good.

Reply:
Also, I don't think it's required to wait for the summary. We should exit as soon as possible, if user asks us to.

Reply:
With utilities like `load-test` and `egress test-template`, the default duration could be a lot longer than the user wants to wait, or they could simply get bored and be happy with a shorter sample of data. It feels like good DX to me to still print this, and it's just a tiny bit of data jockeying and string formatting.

Reply:
And because of the way the hook is unregistered, no matter what happens, sending a second SIGINT or whatever will always insta-terminate

Reply:
True. I think that's a future refactor that requires moving things around a bit. Suffice it to say that this https://github.com/livekit/livekit-cli/pull/343/files#diff-16e3a831829b9fec5dc26d7fa057e9d0326c598b0be63ff443a7794f271e4705L344-R365 combined with using a root NotifyContext, accomplishes the same thing for now.